src/org/rascalmpl/ast/AbstractAST.java

```diff
@@ -69,6 +69,9 @@ public Type getConcreteSyntaxType() {
 	 * Used in generated clone methods to avoid case distinctions in the code generator
 	 */
 	protected <T extends AbstractAST> T clone(T in) {
+	    if (in == null) {
+	        return null;
```

Comment:
this should throw an exception. we don't have null asts, and if we do we have to know about it.

Reply:
maybe better to special case the body of the clone method for the case of optional fields and `null`?

Reply:
my understanding was that this specific function is only called to clone the fields of the AST, and if one of those fields is an optional, that field could have a null value, to avoid the NPE, it should remain null.